packages/rscode/src/extension.ts

```diff
@@ -179,6 +181,80 @@ ${text}
     }
   );
 
+  // Add Docs command
+  const addDocs = vscode.commands.registerTextEditorCommand(
+    'extension.rocksetAdd',
+    async (activeEditor) => {
+      try { // try to parse JSON
+        const docs = JSON.parse(activeEditor.document.getText());
+        try {
+          client.workspaces.listWorkspaces().then(function (rawWorkspaces) { // list workspaces
+            let workspaces: string[] = []
+
+            rawWorkspaces.data?.forEach(item => { // for each workspace, add it's name to `workspaces`
```

Comment:
The other files in the extension use the `map` function. Is `forEach` fine, or should I use `map` for this situation?

Reply:
in this case, since you're putting together a list from a previous list, you should use map. You shouldn't need to type check either, since item.name will always be a string. instead of 
```
let workspaces: string[] = []
rawWorkspaces.data?.forEach(item => {
    if (typeof (item.name) === "string") { workspaces.push(item.name) }
});
```
you can just do:

`const workspaces = rawWorkspaces.map(w => w.name)`